### PR TITLE
[10.x] Cookies Having Independent Partitioned State (CHIPS)

### PR DESCRIPTION
**Pull request by fabricecw**
Implementation of upcoming `partitioned` cookie flag from https://github.com/symfony/symfony/pull/52002.

The modifaction of `Illuminate\Cookie\CookieJar` and `Illuminate\Cookie\CookieServiceProvider` is excluded in this PR since it would require a lot of forward compatibility changes. I would omit these changes in 10.x and extend the signatures with 11.x (possible BC).

Further, there is a separate commit for a cleanup in cookie duplication. Currently, the duplication instantiates a new cookie by passing all attributes again. This is prone to loss of new cookie attributes. It could be simplified to `withValue()` only since this already duplicates the object.

**This PR remains a draft until release of symfony 6.4!**

### CODE REVIEW

**Comment by erikdonohoo** [comment]
With Symony 6.4 releasing today, what do you think the timeline for this will be? We are hoping to use CHIPs in our project soon.

**Comment by fabricecw** [comment]
I will do some tests and undraft it tomorrow. There will be no breaking changes.

**Comment by taylorotwell** [comment]
Tests seem to be failing?

**Comment by fabricecw** [comment]
Github actions aren't retriggered when marking the PR ready for review. I guess I have to ammend a force-push the last commit to trigger the tests again. Local tests are passing.

File: composer.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
         "symfony/console": "^6.2",
         "symfony/error-handler": "^6.2",
         "symfony/finder": "^6.2",
-        "symfony/http-foundation": "^6.3",
+        "symfony/http-foundation": "^6.4",
```
Comment (driesvints) [inline composer.json:47]: This dependency in component `composer.json` files also needs to be bumped.

Comment (fabricecw) [reply composer.json:47]: I noticed `illuminate/session` doesn't use the `illuminate/cookie` dependency. Am I missing something or is it correct to add the dependency?

Comment (fabricecw) [reply composer.json:47]: Sorry, I didn't check it properly. `StartSession` uses the Symfony Cookie. I will bump the version there as well.

Comment (driesvints) [reply composer.json:47]: @fabricecw there's more components that use `symfony/http-foundation`. Can you do a search and bump all of them please?

Comment (fabricecw) [reply composer.json:47]: Absolutely. The other components would't necessarily require 6.4. But I guess it's a good idea to use 6.4 everywhere.

**Pull request merged**